tikv-client/src/main/java/com/pingcap/tikv/TiBatchWrite2PCClient.java

```diff
@@ -0,0 +1,584 @@
+/*
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.pingcap.tikv;
+
+import com.google.protobuf.ByteString;
+import com.pingcap.tikv.exception.GrpcException;
+import com.pingcap.tikv.exception.TiBatchWriteException;
+import com.pingcap.tikv.region.RegionManager;
+import com.pingcap.tikv.region.TiRegion;
+import com.pingcap.tikv.txn.TxnKVClient;
+import com.pingcap.tikv.txn.type.BatchKeys;
+import com.pingcap.tikv.txn.type.ClientRPCResult;
+import com.pingcap.tikv.txn.type.GroupKeyResult;
+import com.pingcap.tikv.util.BackOffFunction;
+import com.pingcap.tikv.util.BackOffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.tikv.kvproto.Kvrpcpb;
+
+public class TiBatchWrite2PCClient {
+
+  public static class ByteWrapper {
+    private byte[] bytes;
+
+    public ByteWrapper(byte[] bytes) {
+      this.bytes = bytes;
+    }
+
+    public byte[] getBytes() {
+      return this.bytes;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      ByteWrapper that = (ByteWrapper) o;
+
+      return Arrays.equals(bytes, that.bytes);
+    }
+
+    @Override
+    public int hashCode() {
+      return Arrays.hashCode(bytes);
+    }
+  }
+
+  public static class BytePairWrapper {
```

Comment:
It can be replaced by guava's Pair.

Reply:
There's an implementation `com.pingcap.tikv.util.Pair`.
In order to use `Pair<F, S>`, we have to put `byte[]` into a Class (e.g. ByteWrapper).
It's a little bit overpackaged.